routs/soldiers.js

```diff
@@ -0,0 +1,29 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, data, done) {
+
+  const parsedUrlString = req.url.split("/");
+  if (parsedUrlString.length >= 3) {
+    done("invalid url request");
```

Comment:
Don't return strings as errors. Use error objects, `new Error(...)`.
Also what you get here is a route which doesn't exist, you should be able to control the error message/code that is returned here, maybe return 404.